### ISSUE TITLE
copied variable names overlap imports

### ISSUE THREAD

**Issue by karlkfi** [maxbrunsfeld/counterfeiter#20]
``` Go
type Registry interface {
    ListComponents(ctx api.Context, labels labels.Selector, fields fields.Selector) (*api.ComponentList, error)
    ...
}
```

``` Go
import (
    "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
    "github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
    "github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
)

...

func (fake *MockRegistry) ListComponents(ctx api.Context, labels labels.Selector, fields fields.Selector) (*api.ComponentList, error) {
    fake.listComponentsMutex.Lock()
    fake.listComponentsArgsForCall = append(fake.listComponentsArgsForCall, struct {
        ctx    api.Context
        labels labels.Selector
        fields fields.Selector
    }{ctx, labels, fields})
    fake.listComponentsMutex.Unlock()
    if fake.ListComponentsStub != nil {
        return fake.ListComponentsStub(ctx, labels, fields)
    } else {
        return fake.listComponentsReturns.result1, fake.listComponentsReturns.result2
    }
}
```

See `labels labels.Selector`.

In this case, labels is the variable name used by the interface definition, which shadows the import. Shadowing is technically fine for usage, as long as the function using it doesn't need to access the import with the same name. Tho it may not pass linting...

**Comment by tjarratt** [comment]
Is this still an issue that you feel strongly about @karlkfi? I'm not aware of anyone that actually lints their counterfeiter fakes, and this change doesn't feel strictly easy for the value it would provide.

How would you feel about closing it? Yay? Nay?

**Comment by karlkfi** [comment]
I'm pretty sure it's actually a compile error when the function wants to use both the variable and the shadowed import. But you might want to confirm.

**Comment by tjarratt** [comment]
Nope, totally compiles just fine, although it is a pretty bad pattern to apply in general.

https://play.golang.org/p/8SVqp5qPTM

I'll leave this open with the label enhancement, it definitely seems like a nice to have since the fake is still functional despite this.

**Comment by joefitzgerald** [comment]
This is fixed in `v6.0.0`, as argument names now use `arg1`, `arg2` names instead of the original names.
